### PR TITLE
add gas tuto

### PR DESCRIPTION
**Pull request by damip**
Explains how gas works in Massa

### CODE REVIEW

File: docs/learn/gas.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+To prevent this, it is required to cap the computation time of each slot below 500 milliseconds to allow the network to follow the slot schedule.
+
+To do this we define a deterministic unit of computation time calibrated on a reference instance of the recommended hardware.
+We call this unit `gas`.
+Each smart contract instruction consumes a defined amount of `gas` when executed.
+
+Protocol-defined gas limits ensuring a total slot execution time remains below 300 milliseconds
+on the reference hardware are set as follows:
+* the execution of autonomous smart contracts at each slot: `MAX_ASYNC_GAS = 1_000_000_000`
+* the execution of all the operations of a block: `MAX_BLOCK_GAS = 1_000_000_000`
```
Comment (AurelienFT) [inline docs/learn/gas.mdx:21]: The value is : 4294967295

**Comment by AurelienFT** [comment]
@Ben-Rey Can you check if you (or someone from your team) understand better and if there is things unclear in this doc please

File: docs/learn/architecture/basic-concepts.mdx [hunk 2]
```diff
@@ -168,6 +168,18 @@ In order to enforce this limit, users are required to lock a corresponding amoun
 Read more about storage costs [here](/docs/learn/storage-costs).
 :::
 
+
+## Gas
+
+In Massa, there is no Gas price.
+Each operation declares a max amount of gas that it can use, and provides a fee that is added to the rewards of the block in which the operation is executed.
+
+Block producers then choose which operations to include in their blocks to fit the max block gas and max block size constraints while maximize the total fee.
```
Comment (qdrn) [inline docs/learn/architecture/basic-concepts.mdx:177]:
```suggestion
Block producers then choose which operations to include in their blocks to fit the max block gas and max block size constraints while maximizing the total fee.
```

**Comment by julienbrs** [comment]
Seems clear to me, however I'm not so sure about a readOnly call, it is still a transaction that we send to a node (right?), but it does approximately 0 computation, so is there a gas amount linked to it ? Does the user need to specify a `max_gas` amount for its readOnly call ?

**Review by qdrn — APPROVED** [review]

**Pull request merged**